plasma_framework/test/src/exits/payment/PaymentInFlightExitable.test.js

```diff
@@ -457,5 +457,220 @@ contract('PaymentInFlightExitable', ([_, alice, bob, carol]) => {
+                    blockNum: txUtxoPos.blockNum,
+                    blockTimestamp: 1000,
+                };
+            }
+
+            function createValidNoncanonicalChallengeArgs(decodedIfeTx) {
+                const utxoPos = new UtxoPos(buildUtxoPos(BLOCK_NUMBER, 2, 2));
+                const { competingTx, decodedCompetingTx, competingTxPos } = createCompetitorTransaction(
+                    decodedIfeTx.inputs[0], utxoPos,
+                );
+
+                const {
+                    competingTxInclusionProof, blockHash, blockNum, blockTimestamp,
+                } = createInclusionProof(
+                    competingTx, competingTxPos,
+                );
+
+                const competingTxWitness = addressToOutputGuard(bob);
+
+                return {
+                    args: {
+                        inFlightTx: web3.utils.bytesToHex(decodedIfeTx.rlpEncoded()),
+                        inFlightTxInputIndex: 0,
+                        competingTx,
+                        competingTxInputIndex: 1,
+                        competingTxInputOutputType: OUTPUT_TYPE_ZERO,
+                        competingTxPos: competingTxPos.utxoPos,
+                        competingTxInclusionProof,
+                        competingTxWitness,
+                    },
+                    block: {
+                        blockHash, blockNum, blockTimestamp,
+                    },
+                    decodedCompetingTx,
+                };
+            }
+
+            beforeEach(async () => {
+                this.framework = await SpyPlasmaFramework.new(
+                    MIN_EXIT_PERIOD, DUMMY_INITIAL_IMMUNE_VAULTS_NUM, INITIAL_IMMUNE_EXIT_GAME_NUM,
+                );
+                this.exitGame = await PaymentInFlightExitable.new(this.framework.address);
+
+                const { args, argsDecoded, inputTxsBlockRoot } = buildValidIfeStartArgs(
+                    AMOUNT, [alice, bob, carol], BLOCK_NUMBER,
+                );
+                this.args = args;
+                this.argsDecoded = argsDecoded;
+                await this.framework.setBlock(BLOCK_NUMBER, inputTxsBlockRoot, 0);
+
+                const conditionTrue = await PaymentSpendingConditionTrue.new();
+
+                await this.exitGame.registerSpendingCondition(
+                    OUTPUT_TYPE_ZERO, IFE_TX_TYPE, conditionTrue.address,
+                );
+
+                await this.exitGame.startInFlightExit(
+                    this.args,
+                    { from: alice, value: IN_FLIGHT_EXIT_BOND },
+                );
+
+                const {
+                    args: cArgs, block, decodedCompetingTx,
+                } = createValidNoncanonicalChallengeArgs(this.argsDecoded.inFlightTx);
+
+                this.challengeArgs = cArgs;
+                this.competingTx = decodedCompetingTx;
+                this.competingTxBlock = block;
+            });
+
+            it('should successfully challenge ife', async () => {
```

Comment:
We don't have test covering the side effect of storage changes.
As a minimum we should add this here. Personally I would nest it with a `describe` and use several `it` for all side effects when challenge successfully to make our test result more explicit.

eg.
```
when successfully challenge inFlight exit
    should emit InFlightExitChallenged event
    should set the oldest competitorPosition
    should set the bond owner to challenger
    should flag the exit non canonical
```